torch/distributed/fsdp/fully_sharded_data_parallel.py

```diff
@@ -1360,6 +1360,25 @@ def _free_full_params_and_use_local_shard(params_to_free):
                         _free_full_params_and_use_local_shard(currently_local_params)
                         self.training_state = TrainingState_.IDLE
 
+    def named_buffers(
+        self,
+        *args,
+        **kwargs,
+    ) -> Iterator[Tuple[str, torch.Tensor]]:
+        """
+        Overrides :meth:`named_buffers()` to intercept buffer names and
+        remove all occurrences of the FSDP-specific flattened buffer prefix
+        when inside the :meth:`summon_full_params` context manager.
+        """
+        in_summon_full_params = getattr(self, "training_state", None) == \
```

Comment:
I'm curious on why can't we access `training_state` directly here.
This property is unconditionally initialized in FSDP's ctor and there's no `del` calls.

Reply:
cc @awgu

Reply:
It is my fault for not digging into this clearly when landing the `named_parameters()` override.

We use this `getattr()` check for `named_parameters()` because the constructor calls `named_parameters()` (twice) before setting `training_state`.
This should not be needed for `named_buffers()` since `named_buffers()` is not called in the constructor.

My thought is that we can either:
- move the definition of `self.training_state = TrainingState_.IDLE` to before the first usage of `named_parameters()` in the constructor, or
- keep `named_parameters()` as is, add a comment explaining why we use `getattr()`, and remove the `getattr()` from `named_buffers()`.

Reply:
https://github.com/pytorch/pytorch/blob/88fd8c8e68e2bbc309f93d0966498c223197c603/torch/distributed/fsdp/fully_sharded_data_parallel.py#L328
https://github.com/pytorch/pytorch/blob/88fd8c8e68e2bbc309f93d0966498c223197c603/torch/distributed/fsdp/fully_sharded_data_parallel.py#L346